docs/layers/tile-layer.md

```diff
@@ -94,32 +94,112 @@ new deck.TileLayer({});
 
 ## Properties
 
-It implements geospatial-specific `tileToBoundingBox` and `getTileIndices` functions as props and also inherits all other [BaseTileLayer](/docs/api-reference/base-tile-layer.md) properties
+### Data Options
 
-### Render Options
-
-##### `getTileData` (Function,  optional)
+##### `getTileData` (Function)
 
 `getTileData` given x, y, z indices of the tile, returns the tile data or a Promise that resolves to the tile data.
 
-- Default: `getTileData: ({x, y, z}) => Promise.resolve(null)`
+- Default: `tile => Promise.resolve(null)`
+
+The `tile` argument contains the following fields:
+
+- `x` (Number) - x index of the tile
+- `y` (Number) - y index of the tile
+- `z` (Number) - z index of the tile
+- `bbox` (Object) - bounding box of the tile, see `tileToBoundingBox`.
+
+By default, the `TileLayer` loads tiles defined by [the OSM tile index](https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames). You may override this by implementing `getTileIndices`.
+
+
+##### `maxZoom` (Number|Null, optional)
+
+Use tiles from this level when over-zoomed.
+
+- Default: `null`
+
+
+##### `minZoom` (Number, optional)
+
+Hide tiles when under-zoomed.
+
+- Default: 0
+
+
+##### `maxCacheSize` (Number|Null, optional)
+
+The maximum cache size for a tile layer. If not defined, it is calculated using the number of tiles in the current viewport times multiplied by `5`.
+
+- Default: `null`
+
+
+##### `strategy` (Enum, optional)
+
+How the tile layer determines the visibility of tiles. One of the following:
+
+* `'best-available'`: If a tile in the current viewport is waiting for its data to load, use cached content from the closest zoom level to fill the empty space. This approach minimizes the visual flashing due to missing content.
+* `'no-overlap'`: Avoid showing overlapping tiles when backfilling with cached content. This is usually favorable when tiles do not have opaque backgrounds.
+
+- Default: `'best-available'`
+
+
+##### `tileToBoundingBox` (Function, optional)
```

Comment:
- I am wondering if there could be a `tileset` prop here, so that the user can pass in a Tileset2D instance?
- And if so, several of the other advanced customization props could be specified (also, or exclusively) on the Tileset2D class
- A lot of the customization for e.g. non-geospatial user cases would happen by users subclassing and redefining Tileset2D
-  If there was sufficient API alignment between tileset2D and tileset3D we might in the future just have a single TileLayer (essentially joining TileLayer and Tile3DLayer). This would of course require a serious generalization effort, so would come later, if at all.

Reply:
+1, I was just hacking around. I wanted to load a static set of tiles at a specific zoom level to work around the low performance of the TerrainLayer and implement a make-shift terrain geofence for a user. Any ideas?